cob_monitoring/src/ping_computers.py

```diff
@@ -0,0 +1,130 @@
+            rospy.logerr("Ping Computers: invalid frequency: %f", diagnostics_frequency)
+            exit            
+        if self.host_pc == "ROBOT":
+            # on robot, publish diagnostics
+            self.diagnostics_publisher = rospy.Publisher('/diagnostics', DiagnosticStatus,queue_size=1)
+            rospy.Timer(rospy.Duration(1.0/diagnostics_frequency), self.publish_diagnostics)
+            # use all but last digits from local IP address
+            ip_prefix = ".".join(subprocess.check_output(['hostname', '-I']).split(".")[:-1])
+            rospy.logdebug("Ping Computers: computer IP: %s, IP prefix: %s",subprocess.check_output(['hostname', '-I']), ip_prefix)
+            for host in self.hosts:
+                host_ip = ip_prefix+"."+str(host)
+                self.hosts_pingable.update({host_ip:False})
+        elif self.host_pc == "ROBOWATCH":
+            # publish directly to topic when running on robowatch pc
+            publish_topic = rospy.get_param('~publish_topic', None)
+            self.slack_publisher = rospy.Publisher(publish_topic, String, queue_size=1)
+            rospy.Timer(rospy.Duration(1.0/diagnostics_frequency), self.publish_slack)
+            # use IP from subnets param
+            subnets = rospy.get_param('~subnets', None)
+            for ip_prefix in subnets:
+                for host in self.hosts:
+                    host_ip = ip_prefix+"."+str(host)
+                    self.hosts_pingable.update({host_ip:False})
+        else:
+            rospy.logerr("Ping Computers: specified host pc not available: got %s, but available are ROBOT and ROBOWATCH", str(self.host_pc))
+            exit
+
+        # ping initially to avoid publishing an error when diagostics_frequency < ping_frequency
+        self.ping_hosts(None)
+        # start timer after all variables have been initialized
+        ping_frequency = rospy.get_param('~ping_frequency', None)
+        if ping_frequency > 0:
+            rospy.Timer(rospy.Duration(1.0/ping_frequency), self.ping_hosts)
+        else:
+            rospy.logerr("Ping Computers: invalid frequency: %f", ping_frequency)
+            exit
+
+    def ping_hosts(self, event):
+        rospy.logdebug("Ping Computers:  hosts to be pinged:\n%s", self.hosts_pingable.keys())
+        for host_ip, pingable in self.hosts_pingable.iteritems():
+            pingable = self.ping(host_ip)
+            rospy.logdebug("Ping Computers: host: %s, pingable %s", str(host_ip), str(pingable))
+            self.hosts_pingable[host_ip] = pingable
+
+    def ping(self, host):
+        """
+        Returns True if host (str) responds to a ping request.
+        """
+        rospy.logdebug("Ping Computers: pinging %s", host)
+        # Building the command. e.g.: "ping -c 1 google.com"
+        command = ['ping', '-c', str(self.ping_number_packages), host, '-q']
+        # if ping returns 0, ping was successful
+        return subprocess.call(command) == 0
```

Comment:
I guess we could get more information out of this and not just true/false, e.g. 
```
ping -c 1 10.4.2.1
PING 10.4.2.1 (10.4.2.1) 56(84) bytes of data.
64 bytes from 10.4.2.1: icmp_seq=1 ttl=63 time=25.2 ms

--- 10.4.2.1 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 25.239/25.239/25.239/0.000 ms
```
if you parse the response of the `subprocess call` then we could put metrics such as package loss and delay into diagnostics

just create one DiagnosticStatus for each pc and append them to `DiagnosticArray`